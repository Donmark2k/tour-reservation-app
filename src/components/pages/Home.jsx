// eslint-disable-next-line import/no-extraneous-dependencies
import { BiRightArrow, BiLeftArrow } from 'react-icons/bi';
import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchTours } from '../../redux/tours/tours';
import { getUser } from '../../util/auth';
import Log from '../Lognin/out/log';

// eslint-disable-next-line consistent-return
// let render = true;
const Home = () => {
  const dispatch = useDispatch();
  const tourS = useSelector((state) => state.tours);
  const { data } = tourS;
  useEffect(() => {
    dispatch(fetchTours());
  }, []);

  const cardsPerPage = 3;
  const totalPages = Math.ceil(data.length / cardsPerPage);

  const [currentPage, setCurrentPage] = useState(1);

  const startIndex = (currentPage - 1) * cardsPerPage;
  const endIndex = startIndex + cardsPerPage;

  const handleNext = () => {
    setCurrentPage((prevPage) => (prevPage < totalPages ? prevPage + 1 : prevPage));
  };

  const handlePrev = () => {
    setCurrentPage((prevPage) => (prevPage > 1 ? prevPage - 1 : prevPage));
  };

  // if (render) {
  //   return (
  //     <div className="flex justify-center items-center h-screen">
  //       <div className="loader" />
  //     </div>
  //   );
  // }
  return (
    <section className="h-screen flex flex-col bg-gray-200">
      <div className="w-full h-16 bg-gray-100 flex justify-between items-center">
        {getUser === null ? <h4 className="ml-12 font-bold text-lg">Welcome</h4>
          : <h4 className="ml-12 font-bold text-lg">{getUser.name}</h4>}
        <Log />
      </div>
      <div className="flex flex-col justify-evenly h-4/5">
        <div className="flex flex-col align-middle text-center">
          <h2 className="text-2xl font-extrabold">LATEST PLACE</h2>
          <h5 className="text-xl text-bGrey">Please Select where you want to visit</h5>
        </div>
        <div className="Home w-full">
          {currentPage > 1 && (
          <button
            type="button"
            onClick={handlePrev}
            disabled={currentPage === 1}
            className="button-pre py-2 px-4 bg-gradient-to-br from-bGreen to-stone-800
                            hover:bg-green-600 text-white font-bold"
          >
            <span><BiLeftArrow /></span>
          </button>
          )}
          <div className="Row">
            {data.length === 0 ? <h3>There is no tour kindly add</h3>
              : data.slice(startIndex, endIndex).map((item) => (
                <div key={item.id} id={item.id} className="card-main hover:w-72 hover:h-72">
                  <a href="/">
                    <img src={`http://localhost:3000${item.image_url}`} alt={item.name} />
                    <div className="leading-4">
                      <h3 className="font-bold text-2xl space-y-1">{item.name}</h3>
                      <p className="text-base">
                        Price:
                        {' '}
                        {item.price}
                        {' '}
                        $
                      </p>
                      <p className="Des text-xs text-bGrey">{item.description}</p>
                    </div>
                  </a>
                </div>
              ))}

          </div>
          {currentPage < totalPages && (
          <button
            type="button"
            onClick={handleNext}
            disabled={currentPage === totalPages}
            className="hover:bg-green-600 font-bold py-2 px-4 text-bGrey bg-gradient-to-br from-green-400 to-slate-600 button-next"
          >
            <span className="text-xl"><BiRightArrow /></span>
          </button>
          )}
        </div>
      </div>
    </section>
  );
};

export default Home;
