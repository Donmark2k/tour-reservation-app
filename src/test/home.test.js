import React from 'react';

import renderer from 'react-test-renderer';

import { BrowserRouter } from 'react-router-dom';

import Home from '../components/pages/Home';

describe('Tests Home component', () => {
  it('Should render Navbar Component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
