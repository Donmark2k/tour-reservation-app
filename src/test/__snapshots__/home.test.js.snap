// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Home component Should render Navbar Component 1`] = `
<section
  className="h-screen flex flex-col justify-evenly items-center  bg-gray-200"
>
  <div
    className="flex flex-col align-middle"
  >
    <h2
      className="text-2xl font-extrabold"
    >
      LATEST PLACE
    </h2>
    <h5
      className="text-xl text-bGrey"
    >
      Please Select where you want to visit
    </h5>
  </div>
  <div
    className="Home "
  >
    <div
      className="Row"
    >
      <div
        className="card-main hover:w-72 hover:h-72"
        id={1}
      >
        <a
          href="/"
        >
          <img
            alt="Tour 1"
            src="https://picsum.photos/200/300"
          />
          <div
            className="leading-4"
          >
            <h3
              className="font-bold text-2xl space-y-1"
            >
              Tour 1
            </h3>
            <p
              className=" text-base"
            >
              Price:
              100
              $
            </p>
            <p
              className="Des text-xs text-bGrey"
            >
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dignissimos, sit.
            </p>
          </div>
        </a>
      </div>
      <div
        className="card-main hover:w-72 hover:h-72"
        id={2}
      >
        <a
          href="/"
        >
          <img
            alt="Tour 2"
            src="https://picsum.photos/200/300"
          />
          <div
            className="leading-4"
          >
            <h3
              className="font-bold text-2xl space-y-1"
            >
              Tour 2
            </h3>
            <p
              className=" text-base"
            >
              Price:
              200
              $
            </p>
            <p
              className="Des text-xs text-bGrey"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quia sed libero facilis fuga, delectus doloribus?
            </p>
          </div>
        </a>
      </div>
      <div
        className="card-main hover:w-72 hover:h-72"
        id={3}
      >
        <a
          href="/"
        >
          <img
            alt="Tour 3"
            src="https://picsum.photos/200/300"
          />
          <div
            className="leading-4"
          >
            <h3
              className="font-bold text-2xl space-y-1"
            >
              Tour 3
            </h3>
            <p
              className=" text-base"
            >
              Price:
              200
              $
            </p>
            <p
              className="Des text-xs text-bGrey"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quia sed libero facilis fuga, delectus doloribus?
            </p>
          </div>
        </a>
      </div>
    </div>
    <button
      className="hover:bg-green-600 font-bold py-2 px-4 text-bGrey bg-gradient-to-br from-green-400 to-slate-600 button-next"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="text-xl"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A.998.998 0 0 0 5 3v18a1 1 0 0 0 .536.886zM7 4.909 17.243 12 7 19.091V4.909z"
          />
        </svg>
      </span>
    </button>
  </div>
</section>
`;

exports[`Tests Navbar component Should render Navbar Component 1`] = `
<section
  className="h-screen flex flex-col justify-evenly items-center  bg-gray-200"
>
  <div
    className="flex flex-col align-middle"
  >
    <h2
      className="text-2xl font-extrabold"
    >
      LATEST PLACE
    </h2>
    <h5
      className="text-xl text-bGrey"
    >
      Please Select where you want to visit
    </h5>
  </div>
  <div
    className="Home "
  >
    <div
      className="Row"
    >
      <div
        className="card-main hover:w-72 hover:h-72"
        id={1}
      >
        <a
          href="/"
        >
          <img
            alt="Tour 1"
            src="https://picsum.photos/200/300"
          />
          <div
            className="leading-4"
          >
            <h3
              className="font-bold text-2xl space-y-1"
            >
              Tour 1
            </h3>
            <p
              className=" text-base"
            >
              Price:
              100
              $
            </p>
            <p
              className="Des text-xs text-bGrey"
            >
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dignissimos, sit.
            </p>
          </div>
        </a>
      </div>
      <div
        className="card-main hover:w-72 hover:h-72"
        id={2}
      >
        <a
          href="/"
        >
          <img
            alt="Tour 2"
            src="https://picsum.photos/200/300"
          />
          <div
            className="leading-4"
          >
            <h3
              className="font-bold text-2xl space-y-1"
            >
              Tour 2
            </h3>
            <p
              className=" text-base"
            >
              Price:
              200
              $
            </p>
            <p
              className="Des text-xs text-bGrey"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quia sed libero facilis fuga, delectus doloribus?
            </p>
          </div>
        </a>
      </div>
      <div
        className="card-main hover:w-72 hover:h-72"
        id={3}
      >
        <a
          href="/"
        >
          <img
            alt="Tour 3"
            src="https://picsum.photos/200/300"
          />
          <div
            className="leading-4"
          >
            <h3
              className="font-bold text-2xl space-y-1"
            >
              Tour 3
            </h3>
            <p
              className=" text-base"
            >
              Price:
              200
              $
            </p>
            <p
              className="Des text-xs text-bGrey"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quia sed libero facilis fuga, delectus doloribus?
            </p>
          </div>
        </a>
      </div>
    </div>
    <button
      className="hover:bg-green-600 font-bold py-2 px-4 text-bGrey bg-gradient-to-br from-green-400 to-slate-600 button-next"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="text-xl"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A.998.998 0 0 0 5 3v18a1 1 0 0 0 .536.886zM7 4.909 17.243 12 7 19.091V4.909z"
          />
        </svg>
      </span>
    </button>
  </div>
</section>
`;
